src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs

```diff
@@ -359,5 +359,257 @@ private static List<SymbolKey> GetOrAddDependentTypes(ConditionalWeakTable<Compi
                             .GetOrAdd(typeId, dependentTypeIds);
             }
         }
+
+        internal static bool OriginalSymbolsMatch(
+            ISymbol searchSymbol,
+            ISymbol symbolToMatch,
+            Solution solution,
+            CancellationToken cancellationToken)
+        {
+            if (ReferenceEquals(searchSymbol, symbolToMatch))
+            {
+                return true;
+            }
+
+            if (searchSymbol == null || symbolToMatch == null)
+            {
+                return false;
+            }
+
+            Compilation symbolToMatchCompilation = null;
+            if (!TryGetCompilation(symbolToMatch, solution, out symbolToMatchCompilation, cancellationToken))
+            {
+                return false;
+            }
+
+            return OriginalSymbolsMatch(searchSymbol, symbolToMatch, solution, null, symbolToMatchCompilation, cancellationToken);
+        }
+
+        internal static bool OriginalSymbolsMatch(
+            ISymbol searchSymbol,
+            ISymbol symbolToMatch,
+            Solution solution,
+            Compilation searchSymbolCompilation,
+            Compilation symbolToMatchCompilation,
+            CancellationToken cancellationToken)
+        {
+            if (symbolToMatch == null)
+            {
+                return false;
+            }
+
+            if (OriginalSymbolsMatchCore(searchSymbol, symbolToMatch, solution, searchSymbolCompilation, symbolToMatchCompilation, cancellationToken))
+            {
+                return true;
+            }
+
+            if (searchSymbol.Kind == SymbolKind.Namespace && symbolToMatch.Kind == SymbolKind.Namespace)
+            {
+                // if one of them is a merged namespace symbol and other one is its constituent namespace symbol, they are equivalent.
+                var namespace1 = (INamespaceSymbol)searchSymbol;
+                var namespace2 = (INamespaceSymbol)symbolToMatch;
+                var namespace1Count = namespace1.ConstituentNamespaces.Length;
+                var namespace2Count = namespace2.ConstituentNamespaces.Length;
+                if (namespace1Count != namespace2Count)
+                {
+                    if ((namespace1Count > 1 &&
+                         namespace1.ConstituentNamespaces.Any(n => NamespaceSymbolsMatch(n, namespace2, solution, cancellationToken))) ||
+                        (namespace2Count > 1 &&
+                         namespace2.ConstituentNamespaces.Any(n2 => NamespaceSymbolsMatch(namespace1, n2, solution, cancellationToken))))
+                    {
+                        return true;
+                    }
+                }
+            }
+
+            if (searchSymbol.Kind == SymbolKind.NamedType && symbolToMatch.IsConstructor())
+            {
+                return OriginalSymbolsMatch(searchSymbol, symbolToMatch.ContainingType, solution, searchSymbolCompilation, symbolToMatchCompilation, cancellationToken);
```

Comment:
When wouldn't a constructor match it's containing type?

This code is a direct copypasta from the other file, but if it's incorrect then we might as well fix it now.